beacon-chain/cache/sync_committee_head_state.go

```diff
@@ -51,13 +50,14 @@ func (c *SyncCommitteeHeadStateCache) Get(slot types.Slot) (state.BeaconState, e
 	if !exists {
 		return nil, ErrNotFound
 	}
-	var st state.BeaconState
-	st, ok := val.(*v2.BeaconState)
+	st, ok := val.(state.BeaconState)
 	if !ok {
-		st, ok = val.(*v3.BeaconState)
-		if !ok {
-			return nil, ErrIncorrectType
-		}
+		return nil, ErrIncorrectType
+	}
+	switch st.Version() {
+	case version.Altair, version.Merge:
+	default:
+		return nil, ErrIncorrectType
 	}
```

Comment:
What is the purpose of this check? Only altair and merge versions work?

Reply:
Yeap, phase0 states do not have sync committees.